src/Datadog.Trace/Util/UriHelpers.cs

```diff
@@ -42,8 +40,48 @@ public static string CleanUriSegment(string absolutePath)
                 return absolutePath;
             }
 
+#if NETCOREAPP
+            ReadOnlySpan<char> absPath = absolutePath.AsSpan();
+
             // Sanitized url will be at worse as long as the original
-            var sb = new StringBuilder(absolutePath.Length);
+            var sb = StringBuilderCache.Acquire(absPath.Length);
+
+            int previousIndex = 0;
+            int index;
+
+            do
```

Comment:
This makes me realize that the loop in the .net framework version should have been a do/while as well

Reply:
Doesn't look like we made this change, @tonyredondo.

Reply:
Oh, you're right, I'll create a new PR with that change.